### DISCUSSION TITLE
[Android SDK] How to make a stream without Activity ?

### DISCUSSION THREAD

**Discussion by toantk238** [ant-media/Ant-Media-Server#6903] [FAQs]
I have used Ant Media Enterprise for Android in many years ( since 2018 ). I just did not updated SDK for 1 year recently.
I just realized that now I cannot start WebRTCClient without a Activity instance.

Please modify the SDK to make the use of Activity instance as optional.

**Comment by Mohit-3196** [comment]
Hi @toantk238,
Let me invite @mustafaboleken to help with this query.

Thank you.

**Reply by rock-tran** [reply]
@Mohit-3196 yes, please.

**Reply by toantk238** [reply]
In my case, I run live stream by using a Foreground Service. Of course, all the required permissions are checked on an activity before.
I understand that you may encourage to use Activity in Pip mode, but I think running as foreground service is second choice.

[My app](https://play.google.com/store/apps/details?id=com.rock.streammapp) is running on Google Play.

**Reply by toantk238** [reply]
@Mohit-3196 Any updates for this ? Google forces to support 16KB page size, which means I need to update Ant Media SDK.

**Reply by amarantmeida** [reply]
@toantk238 We have filed an issue for this and it will be discussed in our sprint meeting today. You can also follow the issue for further updates. 
https://github.com/ant-media/Ant-Media-Server/issues/7483

**Reply by rock-tran** [reply]
@amarantmeida In fact, there are 2 issues that I care 
1. Support page size 16KB
2. Initialize Stream without the need of an Activity instance (Ant Media Android SDK)

**Reply by amarantmeida** [reply]
@rock-tran Like advised earlier, 
#1 is beinf tracked under https://github.com/ant-media/Ant-Media-Server/issues/7483
#2 I am tagging our senior developer @burak-58 to comment on this.

**Reply by toantk238** [reply]
@burak-58 May you make a comment ?
